{{cookiecutter.repo_name}}/install_os_dependencies.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+if [[ $EUID -ne 0 ]]; then
+    echo -e "\nYou must run this with root privilege" 2>&1
+    echo -e "Please do:\n" 2>&1
+    echo "sudo ./${0##*/}" 2>&1
+    echo -e "\n" 2>&1
+
+    exit 1
+else
+
+    apt-get update
+
+    # Install the basic compilation dependencies and other required libraries of this project
+    cat requirements.apt | grep -v "#" | xargs sudo apt-get install -y
```

Comment:
These commands will not only install all packages that are necessary, but will also upgrade any packages already installed to the latest version.  That may not be a bad thing, but I prefer to err on the side of caution when doing root-level administration of somebody else's system.  Having an `--(no-)upgrade` option, or at least warning the user in the docs that packages will be updated would be better.